### PR TITLE
Azure CI: Avoid deprecated `compiler/src/win{32,64}.mak` Makefiles

### PR DESCRIPTION
**Pull request by kinke**
Using `build.d` directly instead, analogous to #15832 for Posix CI.

### CODE REVIEW

File: .azure-pipelines/windows.sh [hunk 1]
```diff
@@ -62,27 +62,36 @@ clone_repos
...
 fi
 
-cd "$DMD_DIR/compiler/src"
-"$DM_MAKE" -f "$MAKE_FILE" MAKE="$DM_MAKE" BUILD=debug unittest
-DFLAGS="-L-LARGEADDRESSAWARE" "$DM_MAKE" -f "$MAKE_FILE" MAKE="$DM_MAKE" GEN="$DMD_DIR\generated" reldmd-asserts
+# avoid the DMC runtime and its limitations for the compiler and {build,run}.d tools themselves
+TOOL_MODEL="$MODEL"
+if [[ "$MODEL" == "32omf" ]]; then
+    TOOL_MODEL=32
+fi
```
Comment (kinke) [inline .azure-pipelines/windows.sh:87]: This translation from `32omf` to `32` for the compiler itself is done in `win32.mak`; I've missed it at first. The accidental first OMF/DMCRT/optlink compiler build immediately crashed with a segfault.

File: .azure-pipelines/windows.sh [hunk 2]
```diff
@@ -62,27 +62,36 @@ clone_repos
...
+
+cd "$DMD_DIR"
+"$HOST_DC" -m$TOOL_MODEL compiler/src/build.d -ofgenerated/build.exe
```
Comment (kinke) [inline .azure-pipelines/windows.sh:90]: Initially I haven't bothered with setting a model for the `build.d` compilation. Then noticed this: https://github.com/dlang/dmd/blob/31c5a894d2680c56870a132a02586403fb9e265a/compiler/src/build.d#L135-L140

File: .azure-pipelines/windows.sh [hunk 3]
```diff
@@ -62,27 +62,36 @@ clone_repos
...
+generated/build.exe -j$N MODEL=$TOOL_MODEL HOST_DMD=$HOST_DC BUILD=debug "${disable_debug_for_unittests[@]}" unittest
+generated/build.exe -j$N MODEL=$TOOL_MODEL HOST_DMD=$HOST_DC DFLAGS="-L-LARGEADDRESSAWARE" ENABLE_RELEASE=1 ENABLE_ASSERTS=1 dmd
```
Comment (kinke) [inline .azure-pipelines/windows.sh:92]: Note: the asserts are currently disabled for Posix CI.

File: .azure-pipelines/windows.sh [hunk 4]
```diff
@@ -102,31 +111,34 @@ cd "$DMD_DIR/../phobos"
...
+    # that, and there's apparently no sane way to specify a libdir for Optlink
+    # in the DMD cmdline either).
     rm "$DMD_DIR/tools/dmd2/windows/bin/sc.ini"
+    # We also need to remove LIB from the freshly built compiler's sc.ini -
+    # not all test invocations use `-conf=`.
+    sed -i 's|^LIB=.*$||g' "$DMD_DIR/generated/windows/release/$TOOL_MODEL/sc.ini"
+    # Okay, now the lib directories are controlled by the LIB env variable.
+    # run.d prepends the dir containing freshly built phobos.lib; we still need
+    # the DMC and Windows libs from the host compiler.
+    export LIB="$DMD_DIR/tools/dmd2/windows/lib"
```
Comment (kinke) [inline .azure-pipelines/windows.sh:141]: OMG, that was a rabbit hole again. In short, I'm not sure how exactly it has been working so far without adapting the generated `sc.ini`, but this was required to get the 32-bit OMF CI job working again when using `build.d` directly.

**Comment by thewilsonator** [comment]
This good to go?

**Review by thewilsonator — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**